### PR TITLE
Add Brotli Support for NGINX in ESP

### PR DESCRIPTION
**Pull request by ilyaGurevich**
Brotli compression is not currently supported in ESP, however the latest version of NGINX found [here](https://nginx.googlesource.com/nginx/+/refs/heads/master/build.bzl#700) indicates that adding a brotli flag to the nginx_repositories function will allow the associated brotli dependencies to be added to ESP. The bazel build of nginx already supports Brotli, all we have to do is just add the flag on the ESP-side.

### CODE REVIEW

**Comment by qiwzhang** [comment]
Are you sure the code is included in the ESP nginx binary?

It seems that you just define the bazel dependency and library target.  you may have to add the lib to [here](https://github.com/cloudendpoints/esp/blob/master/src/nginx/main/ngx_modules.c#L1240)
and [here](https://github.com/cloudendpoints/esp/blob/master/src/nginx/main/BUILD#L102)

**Comment by qiwzhang** [comment]
Please rebase your code

**Comment by qiwzhang** [comment]
What is your plan of adding Brotli compression related config to ESP nginx config?  Do you plan to add some start_esp flags for them?   Or you plan to use your own custom nginx config?

**Comment by qiwzhang** [comment]
We have integration tests written in perl. They are all in src/nginx/t folder.   Could you also add an integration test for this feature.

You can follow this [test](https://github.com/cloudendpoints/esp/blob/master/src/nginx/t/no_service_control.t) 
as example,  its "response2" skip service control,  you just need to compress the response data and verify it

File: src/nginx/BUILD [hunk 1]
```diff
@@ -121,5 +121,7 @@ cc_library(
         "//src/grpc:zero_copy_stream",
         "@nginx//:core",
         "@nginx//:http",
+        "@ngx_brotli//:http_brotli_filter",
+        "@ngx_brotli//:http_brotli_static",
```
Comment (qiwzhang) [inline src/nginx/BUILD:125]: do we need static?  Here is what the doc say

ngx_brotli is a set of two nginx modules:

ngx_brotli filter module - used to compress responses on-the-fly,
ngx_brotli static module - used to serve pre-compressed files.

Do we need o support serve static compressed files.  ESP is just a proxy.

Comment (ilyaGurevich) [reply src/nginx/BUILD:125]: I think I can see your point, but that I was assuming the functionality here is similar to [gzip](http://nginx.org/en/docs/http/ngx_http_gzip_module.html) and [gzip_static](http://nginx.org/en/docs/http/ngx_http_gzip_static_module.html). The latter which appears to be enabled in the nginx_esp binary.. (--with-http_gzip_static_module)

**Comment by ilyaGurevich** [comment]
> What is your plan of adding Brotli compression related config to ESP nginx config? Do you plan to add some start_esp flags for them? Or you plan to use your own custom nginx config?

We will be using our own nginx.conf that includes the brotli directives.

**Comment by qiwzhang** [comment]
LGTM.  I will trigger a presubmit test.

**Pull request merged**